grammar.js

```diff
@@ -51,18 +51,21 @@ module.exports = grammar({
     ),
     classBody: $ => seq("{", optional($.classMembers), "}"),
     classMembers: $ => repeat1($.declaration),
-    declaration: $ => seq(optional($.modifiers), choice(
+    declaration: $ => choice(
       $.fieldDefn,
-    ), optional(";")),
+      // TODO
+    ),
     fieldDefn: $ => seq(
+      optional($.modifiers),
       "var",
       $.id,
       optional(seq(":",
         field("type", $.type))),
       optional(seq("as",
         optional("readonly"), field("as", $.id))),
       optional(seq("=",
-        field("value", $._expression)))
+        field("value", $._expression))),
+      optional(";")
```

Comment:
I'm not 100% sure that this representation is optimal compared to the previous way, where the modifier is located on the declaration, but it is consistent with how this is done in the Java grammar https://github.com/tree-sitter/tree-sitter-java/blob/12c4848bc6cf2660d80b435f8d8de1a9d4676d04/grammar.js#L1015-L1020 and it makes mapping to the AST cleaner. So I suggest we adopt this approach for now.

The same change should probably be made then to move the modifiers under the class definition, rather than having it live under the `start` / `source_file` node.